File: src/2019/day18/rust/benches/bench.rs
#[macro_use]
extern crate criterion;

use aoclib::Solvable;
use criterion::Criterion;

fn part_one_benchmark(c: &mut Criterion) {
	c.bench_function("2019 day 18 part one", |b| {
		let input = aoclib::reader(2019, 18, "input.txt").unwrap();
		b.iter(|| aoc201918::PartOne::solve(&input).unwrap())
	});
}

fn part_two_benchmark(c: &mut Criterion) {
	c.bench_function("2019 day 18 part two", |b| {
		let input = aoclib::reader(2019, 18, "input.txt").unwrap();
		b.iter(|| aoc201918::PartTwo::solve(&input).unwrap())
	});
}

criterion_group!(benches, part_one_benchmark, part_two_benchmark);
criterion_main!(benches);

File: solutions/typescript/2021/09/bench.ts
import { read } from '@lib';
import { defaultBench } from '@lib/benchmark';
import { add } from 'benny';
import { day, year } from '.';
import { runner as partOneRunner } from './part_one';
import { runner as partTwoRunner } from './part_two';

defaultBench(
	'2021 - Day 9',
	add('Part One', async () => {
		const input = (await read(year, day)()).input;
		return () => partOneRunner(input);
	}),
	add('Part Two', async () => {
		const input = (await read(year, day)()).input;
		return () => partTwoRunner(input);
	})
);
